pkg/config/application_test.go

```diff
@@ -132,6 +129,349 @@ func TestNewApplicationGroupSkipAsForeign(t *testing.T) {
+								"value": {
+									"anchor_peers": [
+									{
+									"host": "host2",
+									"port": 123
+									},
+									{
+									"host": "host4",
+									"port": 123
+									}
+									]
+								},
+								"version": "0"
+							},
+							"MSP": {
+								"mod_policy": "Admins",
+								"value": {
+									"config": null,
+									"type": 0
+								},
+								"version": "0"
+							}
+						},
+						"version": "0"
+					}
+				},
+				"mod_policy": "Admins",
+				"policies": {
+					"Admins": {
+						"mod_policy": "Admins",
+						"policy": {
+							"type": 3,
+							"value": {
+								"rule": "MAJORITY",
+								"sub_policy": "Admins"
+							}
+						},
+						"version": "0"
+					},
+					"Readers": {
+						"mod_policy": "Admins",
+						"policy": {
+							"type": 3,
+							"value": {
+								"rule": "ANY",
+								"sub_policy": "Readers"
+							}
+						},
+						"version": "0"
+					},
+					"Writers": {
+						"mod_policy": "Admins",
+						"policy": {
+							"type": 3,
+							"value": {
+								"rule": "ANY",
+								"sub_policy": "Writers"
+							}
+						},
+						"version": "0"
+					}
+				},
+				"values": {
+					"ACLs": {
+						"mod_policy": "Admins",
+						"value": {
+							"acls": {
+								"acl1": {
+									"policy_ref": "hi"
+								}
+							}
+						},
+						"version": "0"
+					},
+					"Capabilities": {
+						"mod_policy": "Admins",
+						"value": {
+							"capabilities": {
+								"V1_3": {}
+							}
+						},
+						"version": "0"
+					}
+				},
+				"version": "0"
+			}
+		},
+		"mod_policy": "",
+		"policies": {},
+		"values": {},
+		"version": "0"
+	},
+	"sequence": "0"
+}
+	`
+
+	expectedUpdatedConfig := &cb.Config{}
+
+	err = protolator.DeepUnmarshalJSON(bytes.NewBufferString(expectedUpdatedConfigJSON), expectedUpdatedConfig)
+	gt.Expect(err).ToNot(HaveOccurred())
+
+	err = AddAnchorPeer(config, "Org1", newOrg1AnchorPeer)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	err = AddAnchorPeer(config, "Org2", newOrg2AnchorPeer)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	gt.Expect(proto.Equal(config, expectedUpdatedConfig)).To(BeTrue())
+}
+
+func TestAddAnchorPeerFailure(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		testName      string
+		orgName       string
+		newAnchorPeer *AnchorPeer
+		expectedErr   string
+	}{
+		{
+			testName:      "When the org for the application does not exist",
+			orgName:       "BadOrg",
+			newAnchorPeer: &AnchorPeer{Host: "host3", Port: 123},
+			expectedErr:   "application org BadOrg does not exist in channel config",
+		},
+		{
+			testName:      "When the anchor peer being added already exists in the org",
+			orgName:       "Org1",
+			newAnchorPeer: &AnchorPeer{Host: "host1", Port: 123},
+			expectedErr:   "application org Org1 already contains anchor peer with host: host1, port: 123",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.testName, func(t *testing.T) {
+			t.Parallel()
```

Comment:
I don't know that the parallelization is buying anything here. Not necessarily opposed to it but we should set our expectations low.

Reply:
I'll leave it in then, we can always revisit if we want to clean up the parallelization and it generally doesn't hurt.